r/R/compute.R

```diff
@@ -186,6 +186,32 @@ unique.ArrowDatum <- function(x, incomparables = FALSE, ...) {
   call_function("unique", x)
 }
 
+#' @export
+any.ArrowDatum <- function(..., na.rm = FALSE){
+  
+  a <- collect_arrays_from_dots(list(...))
+  result <- call_function("any", a)
+
+  if(!as.vector(result) && a$null_count > 0 && !na.rm){
```

Comment:
It would be best to put the `!na.rm` at the beginning here, because it's the only one of these three logicals that does not have any compute cost to calculate. Putting it the beginning allows the expression to short-circuit and never evaluate the other two if `na.rm` is `TRUE`

Reply:
ditto for the same line in the `all` definition below

Reply:
In `scalar_aggregate()` and `median.ArrowDatum`, we check na.rm and null_count before calling the C++ function. Why should any/all do it differently?

Reply:
The case with `any` and `all` is different—it's about three-valued logic, e.g. with the default `na.rm = FALSE`, `any(c(TRUE, NA))` returns `TRUE` but `any(c(FALSE, NA))` returns `NA`

Reply:
Ah right, got it, thanks.

Reply:
Thanks for the suggestion, have updated this now!